pyro/contrib/gp/models/svgp.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Variable
+from torch.distributions import constraints, transform_to
+from torch.nn import Parameter
+
+import pyro
+import pyro.distributions as dist
+from pyro.distributions.util import matrix_triangular_solve_compat
+
+from .vgp import VariationalGP
+
+
+class SparseVariationalGP(VariationalGP):
```

Comment:
isn't your class hierarchy backwards here?

Reply:
Yes, although VariationGP is a special case of sparse version (`Xu = X`), there is a different point (`Xu` is a parameter in sparse version). We can think that Sparse method is an **extended** version of VariationGP (by introducing a new parameter `Xu`). I found that it is simpler to implement this way. Either way, we have to rewrite `model` and `guide`.

Reply:
did you end up with much of any actual code reuse?